src/components/tasks/Multichoice.vue

```diff
@@ -0,0 +1,112 @@
+  import { useAuthStore } from '@/store/auth';
+  import { useGameStore } from '@/store/game';
+  import _head from 'lodash/head';
+  import _get from 'lodash/get';
+  
+  const taskId = "multichoice"
+  const router = useRouter();
+  const gameStarted = ref(false);
+  const authStore = useAuthStore();
+  const gameStore = useGameStore();
+  const { roarfirekit, firekitUserData, isFirekitInit } = storeToRefs(authStore);
+  
+  // Send user back to Home if page is reloaded
+  const entries = performance.getEntriesByType("navigation");
+  entries.forEach((entry) => {
+    if (entry.type === "reload") {
+      // Detect if our previous reload was on this page, AND if the last naviagtion was a replace.
+      if (entry.name === window.location.href && history.state.replaced === true) {
+        router.replace({ name: "Home" })
+      }
+    }
+  });
+  
+  // The following code intercepts the back button and instead forces a refresh.
+  // We use the ``preventBack`` variable to prevent an infinite loop. I.e., we
+  // only want to intercept this the first time.
+  let preventBack = true;
+  onBeforeRouteLeave((to, from, next) => {
+    if (window.event.type === "popstate" && preventBack) {
+      preventBack = false;
+      // router.go(router.currentRoute);
+      router.go(0);
+    } else {
+      next();
+    }
+  });
+  
+  onMounted(async () => {
+    if (isFirekitInit.value) {
+      await startTask();
+    }
+  })
+  
+  watch(isFirekitInit, async (newValue, oldValue) => {
+    await startTask();
+  })
+  
+  let roarApp;
+  
+  const completed = ref(false);
+  const { selectedAdmin } = storeToRefs(gameStore);
+  
+  const selectBestRun = async () => {
+    await authStore.roarfirekit.selectBestRun({
+      assignmentId: selectedAdmin.value,
+      taskId,
+    })
+  }
+  
+  window.addEventListener('beforeunload', selectBestRun, { once: true });
+  onBeforeUnmount(async () => {
+    if (roarApp && completed.value === false) {
+      roarApp.abort();
+    }
+    selectBestRun();
+  });
+  
+  async function startTask() {
+    const appKit = await authStore.roarfirekit.startAssessment(selectedAdmin.value, taskId)
+  
+    const userDob = _get(roarfirekit.value, 'userData.studentData.dob') || _get(firekitUserData.value, 'studentData.dob')
+    const userDateObj = new Date(toRaw(userDob).seconds * 1000)
+  
+    const userParams = {
+      birthMonth: userDateObj.getMonth() + 1,
+      birthYear: userDateObj.getFullYear(),
+    }
+  
+    const gameParams = appKit._taskInfo.variantParams
+    roarApp = new RoarMultichoice(appKit, gameParams, userParams, 'jspsych-target');
+  
+    gameStarted.value = true;
+    await roarApp.run().then(async () => {
+      // Handle any post-game actions.
+      await authStore.roarfirekit.completeAssessment(selectedAdmin.value, taskId)
+      router.replace({ name: "Home" });
+    });
+  }
+  </script>
+  <style scoped> .game-target {
+     position: absolute;
+     top: 0;
+     left: 0;
+     width: 100%;
```

Comment:
```suggestion
     width: 100%;
     height: 100%;
```
Incorporate the game freeze css fix. Can you also diff this component against the latest game components to ensure that it incorporates any changes that have happened to them since you opened the PR?